lib/elixir_sense/core/state.ex

```diff
@@ -188,24 +190,36 @@ defmodule ElixirSense.Core.State do
   def escape_protocol_impementations({protocol, implementations}) do
     joined_implementations =
       implementations
-      |> Enum.map(fn parts ->
-        parts
-        |> Enum.map(&Atom.to_string/1)
-        |> Enum.join(@dot_marker)
+      |> Enum.map(fn
+        parts when is_list(parts) ->
+          parts
+          |> Enum.map(&Atom.to_string/1)
+          |> Enum.join(@dot_marker)
+
+        module when is_atom(module) ->
+          Atom.to_string(module) |> String.replace("Elixir.", "")
       end)
       |> Enum.join(@or_marker)
       |> String.to_atom()
 
-    protocol ++ [joined_implementations]
+    if is_list(protocol) do
```

Comment:
You could use `List.wrap(protocol) ++ [joined_implementations]` because it already does this logic for you